eskipfile/remote.go

```diff
@@ -154,31 +161,42 @@ func isFileRemote(remotePath string) bool {
 func (client *remoteEskipFile) DownloadRemoteFile() error {
 	data, err := client.getRemoteData()
 	if err != nil {
+		if errors.Is(err, errContentNotChanged) {
+			return nil
+		}
 		return err
 	}
-	defer data.Close()
 
-	out, err := os.OpenFile(client.localPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	err = os.WriteFile(client.localPath, data, 0600)
+	return err
+}
+
+func (client *remoteEskipFile) getRemoteData() ([]byte, error) {
+	req, err := http.NewRequest("GET", client.remotePath, nil)
+
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	if _, err = io.Copy(out, data); err != nil {
-		_ = out.Close()
-		return err
+	if client.etag != "" {
+		req.Header.Set("If-None-Match", client.etag)
 	}
 
-	return out.Close()
-}
-
-func (client *remoteEskipFile) getRemoteData() (io.ReadCloser, error) {
-	resp, err := client.http.Get(client.remotePath)
+	resp, err := client.http.Do(req)
 	if err != nil {
 		return nil, err
 	}
+	defer resp.Body.Close()
+
+	if client.etag != "" && resp.StatusCode == 304 {
+		return nil, errContentNotChanged
+	}
+
 	if resp.StatusCode != 200 {
-		return nil, errors.New("download file failed")
+		return nil, fmt.Errorf("failed to download remote file %s, status code: %d", client.remotePath, resp.StatusCode)
 	}
 
-	return resp.Body, nil
+	client.etag = resp.Header.Get("ETag")
+
+	return io.ReadAll(resp.Body)
```

Comment:
This reads everything into memory just to dump into the file a moment later. Previuosly it was correctly returning io.ReadCloser